src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,403 @@
+        if (key == null) {
+            return null;
+        }
+        return hexEncode(getKeyDigestMD5(key));
+    }
+
+    /**
+     * Generates an SHA1 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigestSHA1(@Nonnull Key k) throws IOException {
+        try {
+            return getKeyDigest(k, "SHA1");
+        } catch (NoSuchAlgorithmException e) {
+           throw new AssertionError("SHA1 algorithm not found to create digest");
+        }
+    }
+
+    /**
+     * Generates an MD5 digest from a Key object
+     * 
+     * @param k the key to generate the digest from
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     */
+    @Nonnull
+    public static byte[] getKeyDigestMD5(@Nonnull Key k) throws IOException {
+        try {
+            return getKeyDigest(k, "MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new AssertionError("SHA1 algorithm not found to create digest");
+        }
+    }
+
+    /**
+     * Generates an digest from a Key object in the specified digest format. The supported digest formats will depend on
+     * the JVM API.
+     * 
+     * @param k key to generate the digest from
+     * @param algorithm digest format
+     * @return the generated digest
+     * @throws IOException if a problem exists creating the digest
+     * @throws NoSuchAlgorithmException when provided digest algorithm is not available
+     */
+    @Nonnull
+    public static byte[] getKeyDigest(@Nonnull Key k, @Nonnull String algorithm) throws IOException, NoSuchAlgorithmException {
+        MessageDigest md5 = MessageDigest.getInstance(algorithm);
+
+        DigestInputStream in = new DigestInputStream(new ByteArrayInputStream(k.getEncoded()), md5);
+        try {
+            while (in.read(new byte[128]) > 0)
+                ; // simply discard the input
+        } finally {
+            in.close();
+        }
+        return md5.digest();
+    }
+
+    /**
+     * Encode {@link byte[]} in hex formated string "ab:cd:ef:...:12"
+     * 
+     * @param data to be encoded
+     * @return hex formated string "ab:cd:ef:...:12"
+     */
+    @Nonnull
+    public static String hexEncode(@Nonnull byte[] data) {
+        char[] hex = Hex.encodeHex(data);
+        StringBuilder buf = new StringBuilder();
+        for (int i = 0; i < hex.length; i += 2) {
+            if (buf.length() > 0)
+                buf.append(':');
+            buf.append(hex, i, 2);
+        }
+        return buf.toString();
+    }
+
+    /**
+     * Encodes a {@link byte[]} in base 64 string
+     * 
+     * @param data to be encoded
+     * @return base 64 formatted string
+     */
+    @Nonnull
+    public static String encodeBase64(@Nonnull byte[] data) {
```

Comment:
:bug: do not expose more than we need to. Base64 encoding/decoding is not a responsibility of this API contract. By making these methods public we expose additional surface to the API and restrict the ability to refactor later